source/JsMaterialX/JsMaterialXCore/post.js

```diff
@@ -0,0 +1,10 @@
+Module.onRuntimeInitialized = function() {
+  Module.ValueElement.prototype.getValue = function () {
+    const value = Module.ValueElement.prototype._getValue.apply(this);
+    if (value) {
+      return value.getData();
```

Comment:
You mentioned this decision last week. Could you elaborate on why we want to diverge from the C++ API here?

Reply:
If it's only for the convenience of having a short-hand for `element.getValue().getData()`, maybe we could call the function `getValueData` instead, and keep the original `getValue`?

Reply:
Well, this is following what the python bindings are doing. If you dont do `getData` you get from `getValue` something like TypedBoolean instead of the value itself. To be honest, I'm not sure exactly how the python bindings work internally but they do the same logic: https://github.com/autodesk-forks/MaterialX/blob/adsk_contrib/dev/python/MaterialX/datatype.py#L67

I just don't see it explicitly in the bindings

Reply:
Hm, the link you shared is a bit confusing. I don't see anything related to `getValue` there. Maybe this is what you wanted to link? https://github.com/autodesk-forks/MaterialX/blob/adsk_contrib/dev/python/MaterialX/main.py#L72

Wouldn't we consequently need to go through all the python bindings and implement methods accordingly, then? `setValue` and `getDefaultValue` seem to have a custom implementation as well in python. And so do many other functions, e.g. `InterfaceElement::getInputValue`. I would propose to discuss this today in our sync up with @bernardkwok and @ashwinbhat.

Reply:
Yeap, sorry. That one I shared does the same thing, but my bad, it was from a different function